src/coreclr/vm/ceeload.cpp

```diff
@@ -4615,16 +4615,16 @@ void Module::RunEagerFixups()
     {
         // For composite images, multiple modules may request initializing eager fixups
         // from multiple threads so we need to lock their resolution.
-        if (compositeNativeImage->EagerFixupsHaveRun())
-        {
-            return;
-        }
         CrstHolder compositeEagerFixups(compositeNativeImage->EagerFixupsLock());
         if (compositeNativeImage->EagerFixupsHaveRun())
         {
+            if (compositeNativeImage->ReadyToRunCodeDisabled())
+                GetReadyToRunInfo()->DisableAllR2RCode();
```

Comment:
So this is basically just propagating up the information from `compositeNativeImage`, is that right? The issue being that we weren't before and so `compositeNativeImage` might have flagged itself as not being R2R compatible but the corresponding module loaded in the VM wouldn't track that?

Reply:
The issue was that the first module in the composite image would trigger the code to be disabled, but the second module and subsequent modules would just use the code in the composite image.